From: Dan Williams <dcbw@redhat.com>
Date: Wed, 20 Jan 2016 10:53:54 -0600
Subject: libnm,vpn: restore export of deprecated NMVpnPluginOld symbols

Deprecated of course, but shouldn't have been removed from the ABI.

Fixes: 867227dd4aac8371b3cacc366b32a3fa96eba42a

(cherry-picked from commit d59c1d4c8aee9f3bee7e30eb7505aace62572fff)
---
 docs/libnm/Makefile.am    |  1 -
 libnm/libnm.ver           | 11 +++++++++++
 libnm/nm-vpn-plugin-old.c | 40 ++++++++++++++++++++++++++++++++++++++++
 3 files changed, 51 insertions(+), 1 deletion(-)

diff --git a/docs/libnm/Makefile.am b/docs/libnm/Makefile.am
index 5b96d5a..380371e 100644
--- a/docs/libnm/Makefile.am
+++ b/docs/libnm/Makefile.am
@@ -54,7 +54,6 @@ IGNORE_HFILES= \
 	nm-setting-private.h \
 	nm-types.h \
 	nm-utils-private.h \
-	nm-vpn-plugin-old.h \
 	nm-vpn-service-plugin.h \
 	nm-core-tests-enum-types.h
 
diff --git a/libnm/libnm.ver b/libnm/libnm.ver
index ed008ee..1b18560 100644
--- a/libnm/libnm.ver
+++ b/libnm/libnm.ver
@@ -821,6 +821,17 @@ global:
 	nm_vpn_plugin_error_get_type;
 	nm_vpn_plugin_error_quark;
 	nm_vpn_plugin_failure_get_type;
+	nm_vpn_plugin_old_disconnect;
+	nm_vpn_plugin_old_failure;
+	nm_vpn_plugin_old_get_connection;
+	nm_vpn_plugin_old_get_secret_flags;
+	nm_vpn_plugin_old_get_state;
+	nm_vpn_plugin_old_get_type;
+	nm_vpn_plugin_old_read_vpn_details;
+	nm_vpn_plugin_old_secrets_required;
+	nm_vpn_plugin_old_set_ip4_config;
+	nm_vpn_plugin_old_set_login_banner;
+	nm_vpn_plugin_old_set_state;
 	nm_vpn_service_state_get_type;
 	nm_wep_key_type_get_type;
 	nm_wimax_nsp_connection_valid;
diff --git a/libnm/nm-vpn-plugin-old.c b/libnm/nm-vpn-plugin-old.c
index cfc8c66..4854b66 100644
--- a/libnm/nm-vpn-plugin-old.c
+++ b/libnm/nm-vpn-plugin-old.c
@@ -129,6 +129,11 @@ nm_vpn_plugin_old_get_connection (NMVpnPluginOld *plugin)
 	return connection;
 }
 
+/**
+ * nm_vpn_plugin_old_get_state:
+ *
+ * Deprecated: 1.2: replaced by NMVpnServicePlugin
+ */
 NMVpnServiceState
 nm_vpn_plugin_old_get_state (NMVpnPluginOld *plugin)
 {
@@ -137,6 +142,11 @@ nm_vpn_plugin_old_get_state (NMVpnPluginOld *plugin)
 	return NM_VPN_PLUGIN_OLD_GET_PRIVATE (plugin)->state;
 }
 
+/**
+ * nm_vpn_plugin_old_set_state:
+ *
+ * Deprecated: 1.2: replaced by NMVpnServicePlugin
+ */
 void
 nm_vpn_plugin_old_set_state (NMVpnPluginOld *plugin,
                              NMVpnServiceState state)
@@ -152,6 +162,11 @@ nm_vpn_plugin_old_set_state (NMVpnPluginOld *plugin,
 	}
 }
 
+/**
+ * nm_vpn_plugin_old_set_login_banner:
+ *
+ * Deprecated: 1.2: replaced by NMVpnServicePlugin
+ */
 void
 nm_vpn_plugin_old_set_login_banner (NMVpnPluginOld *plugin,
                                     const char *banner)
@@ -162,6 +177,11 @@ nm_vpn_plugin_old_set_login_banner (NMVpnPluginOld *plugin,
 	g_signal_emit (plugin, signals[LOGIN_BANNER], 0, banner);
 }
 
+/**
+ * nm_vpn_plugin_old_failure:
+ *
+ * Deprecated: 1.2: replaced by NMVpnServicePlugin
+ */
 void
 nm_vpn_plugin_old_failure (NMVpnPluginOld *plugin,
                            NMVpnPluginFailure reason)
@@ -171,6 +191,11 @@ nm_vpn_plugin_old_failure (NMVpnPluginOld *plugin,
 	g_signal_emit (plugin, signals[FAILURE], 0, reason);
 }
 
+/**
+ * nm_vpn_plugin_old_disconnect:
+ *
+ * Deprecated: 1.2: replaced by NMVpnServicePlugin
+ */
 gboolean
 nm_vpn_plugin_old_disconnect (NMVpnPluginOld *plugin, GError **err)
 {
@@ -277,6 +302,11 @@ schedule_fail_stop (NMVpnPluginOld *plugin)
 	priv->fail_stop_id = g_idle_add (fail_stop, plugin);
 }
 
+/**
+ * nm_vpn_plugin_old_set_config:
+ *
+ * Deprecated: 1.2: replaced by NMVpnServicePlugin
+ */
 void
 nm_vpn_plugin_old_set_config (NMVpnPluginOld *plugin,
                               GVariant *config)
@@ -316,6 +346,11 @@ nm_vpn_plugin_old_set_config (NMVpnPluginOld *plugin,
 	g_signal_emit (plugin, signals[CONFIG], 0, config);
 }
 
+/**
+ * nm_vpn_plugin_old_set_ip4_config:
+ *
+ * Deprecated: 1.2: replaced by NMVpnServicePlugin
+ */
 void
 nm_vpn_plugin_old_set_ip4_config (NMVpnPluginOld *plugin,
                                   GVariant *ip4_config)
@@ -371,6 +406,11 @@ nm_vpn_plugin_old_set_ip4_config (NMVpnPluginOld *plugin,
 		nm_vpn_plugin_old_set_state (plugin, NM_VPN_SERVICE_STATE_STARTED);
 }
 
+/**
+ * nm_vpn_plugin_old_set_ip6_config:
+ *
+ * Deprecated: 1.2: replaced by NMVpnServicePlugin
+ */
 void
 nm_vpn_plugin_old_set_ip6_config (NMVpnPluginOld *plugin,
                                   GVariant *ip6_config)
